### PR TITLE
Add the ability to filter tags by the existence of a field

### PR DESCRIPTION
**Pull request by currysoup**
Currently only support contributorInfromation.bylineImage and ".largeBylineImage since this is all that's needed for now.

### CODE REVIEW

File: app/repositories/TagRepository.scala [hunk 1]
```diff
@@ -108,6 +110,17 @@ case class TagSearchCriteria(
 
   private def typeFilter(types: List[String])(tags: List[Tag]) = tags.filter { t => types.contains(t.`type`.toLowerCase) }
 
+  private def hasFieldsFilter(fields: List[String])(tags: List[Tag]) = tags.filter { t =>
+    fields.forall {
+        case "contributorinformation.bylineimage" => t.contributorInformation.flatMap(_.bylineImage).isDefined
+        case "contributorinformation.largebylineimage" => t.contributorInformation.flatMap(_.largeBylineImage).isDefined
+        case unsupported => {
+          Logger.warn(s"Attempted to check if tag has field '$unsupported' which is not supported by the hasFieldsFilter")
```
Comment (mbarton) [inline app/repositories/TagRepository.scala]: Could we communicate this failure back to the client?

File: app/controllers/TagManagementApi.scala [hunk 2]
```diff
@@ -51,12 +51,13 @@ object TagManagementApi extends Controller with PanDomainAuthActions {
...
   def searchTags = APIAuthAction { req =>
-    
+
     val criteria = TagSearchCriteria(
       q = req.getQueryString("q"),
       searchField = req.getQueryString("searchField"),
       types = req.getQueryString("types").map(_.split(",").toList),
-      referenceType = req.getQueryString("referenceType")
+      referenceType = req.getQueryString("referenceType"),
+      hasFields = req.getQueryString("hasFields").map(_.split(",").toList)
```
Comment (mbarton) [inline app/controllers/TagManagementApi.scala]: Potentially needs a `trim`?

Comment (currysoup) [reply app/controllers/TagManagementApi.scala]: Sure, I'll add that to `hasFields` and `types`!

File: app/repositories/TagRepository.scala (hunk 1 above)
Comment (currysoup) [reply app/repositories/TagRepository.scala]: Was previously just defaulted to true if it was unsupported... might just go back to that.

**Pull request merged**